python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -215,6 +220,83 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     return embedding
 
 
+def categorical_embedding_with_vocabulary_file_and_dropout(
+    feature_tensor, feature_info, file_io: FileIO
+):
+    """
+    Converts a string tensor into a categorical embedding representation.
+    Works by using a vocabulary file to convert the string tensor into categorical indices
+    and then converting the categories into embeddings based on the feature_info.
+    Also uses a dropout to convert categorical indices to the OOV index of 0 at a rate of dropout_rate
+
+    Args:
+        feature_tensor: String feature tensor
+        feature_info: Dictionary representing the configuration parameters for the specific feature from the FeatureConfig
+
+    Returns:
+        Categorical embedding representation of input feature_tensor
+
+    Args under feature_layer_info:
+        vocabulary_file: str; path to vocabulary CSV file for the input tensor
+        dropout_rate: float; rate at which to convert categorical indices to OOV
+        embedding_size: int; dimension size of categorical embedding
+
+    NOTE:
+    The vocabulary CSV file must contain two columns - key, id,
+    where the key is mapped to one id thereby resulting in a
+    many-to-one vocabulary mapping.
+    If id field is absent, a unique natural number id is assigned by default
+    resulting in a one-to-one mapping
+
+    OOV index will be set to 0
+    num_oov_buckets will be 0
+    """
+    feature_layer_info = feature_info.get("feature_layer_info")
+    vocabulary_df = file_io.read_df(feature_layer_info["args"]["vocabulary_file"])
```

Comment:
Why can't we just call ml4ir/base/features/feature_fns/categorical.py#get_vocabulary_info() or adapt it if needed?

Reply:
Yeah. I will do that. I'm focusing on training models first. Will address this next week if that is okay with you.

Reply:
What the latest on this?

Reply:
Fixed.